tests/unit/test_config.py

```diff
@@ -89,6 +93,109 @@ def test_can_chain_chalice_stage_values():
     assert not prod.manage_iam_role
 
 
+def test_can_chain_function_values():
+    disk_config = {
+        'lambda_timeout': 10,
+        'stages': {
+            'dev': {
+                'lambda_timeout': 20,
+                'lambda_functions': {
+                    'api_handler': {
+                        'lambda_timeout': 30,
+                    }
+                }
+            }
+        }
+    }
+    c = Config(chalice_stage='dev',
+               config_from_disk=disk_config)
+    assert c.lambda_timeout == 30
+
+
+def test_can_create_scope_obj_with_new_function():
+    disk_config = {
+        'lambda_timeout': 10,
+        'stages': {
+            'dev': {
+                'lambda_timeout': 20,
+                'lambda_functions': {
+                    'api_handler': {
+                        'lambda_timeout': 30,
+                    },
+                    'myauth': {
+                        'lambda_timeout': 40,
+                    }
+                }
+            }
+        }
+    }
+    c = Config(chalice_stage='dev', config_from_disk=disk_config)
+    new_config = c.scope(chalice_stage='dev',
+                         function_name='myauth')
+    assert new_config.lambda_timeout == 40
```

Comment:
So I know it may be tedious, but I wonder if it makes sense to have explicit test to make sure all the valid variables can be configured per lambda function? In the tests, `lambda_timeout`, `environment_variables`, and `tags` are used routinely, but variables such as `iam_role_arn`, `lambda_memory_size`, `autogen_policy`, and `iam_policy_file` are not. I feel that it would be good to have for a quick check. If not split this into individual tests, maybe have one large test or general helper assertion that makes sure that each of these values exhibit the per function behavior (i.e. override global/stage values and are separate from other functions)?